apps/total-typescript/src/server/get-middleware-response.ts

```diff
@@ -42,6 +46,18 @@ export async function getMiddlewareResponse(req: NextRequest) {
   //   }
   // }
 
+  if (req.nextUrl.pathname.includes('/admin')) {
+    try {
+      const user = UserSchema.parse(token)
+      const ability = getCurrentAbility({user})
+      if (!ability.can('create', 'Content')) {
+        response = rewriteToPath('/login', req)
+      }
+    } catch (error) {
+      response = rewriteToPath('/login', req)
+    }
+  }
```

Comment:
this is guarding the /admin route

Reply:
I haven't used middleware in Next before so it's neat to see it in action